### ISSUE TITLE
Function `split_nth` throws away the nth character.

### ISSUE THREAD

**Issue by katekyy** [vlang/v#18997]
### Describe the bug

The `.split_nth(delim, nth)` function is throwing away character at index `nth`.

For example:
```v
'this is a very long string of... characters. its surely longer than 16 bytes long.'.split_nth('', 17)
```
Returns:
```v
['t', 'h', 'i', 's', ' ', 'i', 's', ' ', 'a', ' ', 'v', 'e', 'r', 'y', ' ', 'l', 'ng string of... characters. its surely longer than 16 bytes long.']
                                                                          //   ^^^
                                                                          // missing "o"
```

### Expected Behavior

I expected the function to not throw away any characters.

### Current Behavior

As said above in the description.

### Reproduction Steps

```v
'A test string tHat will... test this.'.split_nth('', 16)
```

### Possible Solution

Workaround:
```v
mut strings := []string{}
x := 'A test string tHat will... test this.'.bytes()

head := x[..16]
tail := x[16..]

strings << head.map(it.ascii_str())
strings << tail.bytestr()
```

### V version

V 0.4.0 2f2dde8 

### Environment details (OS name and version, etc.)

V full version: V 0.4.0 e7af25e.2f2dde8
OS: windows, Microsoft Windows 10 Pro v19045 64-bit
Processor: 8 cpus, 64bit, little endian,

getwd: C:\\.programming\v
vexe: C:\v\v.exe
vexe mtime: 2023-07-28 12:44:32

vroot: OK, value: C:\v
VMODULES: OK, value: C:\Users\katekyy\.vmodules
VTMP: OK, value: C:\Users\katekyy\AppData\Local\Temp\v_0

Git version: git version 2.38.1.windows.1
Git vroot status: weekly.2023.28-108-g2f2dde8a
.git/config present: true

CC version: Error: exec failed (CreateProcess) with code 2: The specified file could not be found.
 cmd: cc --version
thirdparty/tcc: N/A

**Comment by katekyy** [comment]
I think I found something:
https://vosca.dev/p/f16683d2f0

**Comment by katekyy** [comment]
I found something https://vosca.dev/p/0596a4455b. 
This executes without errors and I don't think it has any bugs.
`v/vlib/builtin/string_test.v` runs it ok too.

**Comment by katekyy** [comment]
I found something https://vosca.dev/p/0596a4455b. 
This executes without errors and I don't think it has any bugs.
`v/vlib/builtin/string_test.v` runs it ok too.

**Comment by katekyy** [comment]
Fixed in #19005.
